indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -496,4 +496,8 @@ private String extractDateYear(Date date) {
         calendar.setTime(date);
         return Integer.toString(calendar.get(Calendar.YEAR));
     }
+
+    private boolean hasNodeValue(Element node) {
```

Comment:
Another approach could be to have a private `retrieveNamePartText(Element nameParts, String fieldName)` method to replace all the `JDOMQueryUtil.getElementByAttribute` calls, which would retrieve the value when the element is present, and then you could switch all your conditionals back to just doing `if (datePart != null)`

Something like
```
Element namePart = JDOMQueryUtil.getElementByAttribute(nameParts, "type", fieldName);
if (namePart == null) {
    return null;
} else {
    String value = namePart.getTextTrim();
    return StringUtils.isBlank(value) ? null : value;
}
```

If you don't want to go with that approach, you should switch to using getTextTrim instead of getValue in this method and everywhere in `formatName`, and line 460 isn't using this helper method.